pkg/logs/sender/message_buffer.go

```diff
@@ -50,7 +51,19 @@ func (p *MessageBuffer) GetMessages() []*message.Message {
 
 // IsFull returns true if the buffer is full.
 func (p *MessageBuffer) IsFull() bool {
-	return len(p.messageBuffer) == cap(p.messageBuffer) || p.contentSize == p.contentSizeLimit
+	return p.MessagesCountReached() || p.ContentSizeLimitReached()
+}
+
+// MessagesCountReached return true if the buffer is full because of how many
+// messages it contains.
+func (p *MessageBuffer) MessagesCountReached() bool {
+	return len(p.messageBuffer) == cap(p.messageBuffer)
+}
+
+// ContentSizeLimitReached returns true if the buffer is full because its content
+// size limit has been reached.
+func (p *MessageBuffer) ContentSizeLimitReached() bool {
+	return p.contentSize == p.contentSizeLimit
```

Comment:
Will this always be equal, though?

Is there no chance that we may never reach the size limit (because we refuse to batch another message if `p.contentSize + len(new_message) > p.contentSizeLimit`? Or do we add messages as long as there's _any_ free space, such that `p.contentSize` might be _greater_ than `p.contentSizeLimit`?

Reply:
Yeah it's incorrect, let me change this. Thank you for spotting that